AzureCalling/Controllers/LobbyViewController.swift

```diff
@@ -124,6 +125,7 @@ class LobbyViewController: UIViewController, UITextFieldDelegate {
 
     private func openAudioDeviceDrawer() {
         let audioDeviceSelectionDataSource = AudioDeviceSelectionDataSource()
+        audioDeviceSelectionDataSource.audioDeviceSelectionDelegate = self
```

Comment:
I think it's more consistent with the original pattern by using a closure here. 
similar to `dismissDrawer` inside `AudioDeviceSelectionDataSource`, we could have a closure which is defined as 
`var didSelectedItem: (BottomDrawerItem) -> void`  using initializer to define the closure
Inside this closure,  we update the icon on main thread
```
audioDeviceSelectionDataSource.didSelectedItem = {[weak self] item
guard let self = self else {return }
updateAudioDeviceButtonIcon(item.avatar)
}
```

Reply:
@AmyL219 Thank you for the feedback and suggestions. Using a closure was a much better approach, trimmed down the amount of codes written. PR now updated